src/routes/v2/authenticated/review.ts

```diff
@@ -967,6 +969,101 @@ export class ReviewsRouter {
     return res.status(200).send()
   }
 
+  getBlob: RequestHandler<
+    { siteName: string; requestId: number },
+    BlobDiffDto | ResponseErrorBody,
+    unknown,
+    { path: string },
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    // Step 1: Check that the site exists
+    const { siteName, requestId } = req.params
+    const { path } = req.query
+    const { userWithSiteSessionData } = res.locals
+    const site = await this.sitesService.getBySiteName(siteName)
+
+    if (!site) {
+      logger.error({
+        message: "Invalid site requested",
+        method: "getBlob",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+        },
+      })
+      return res.status(404).send({
+        message: "Please ensure that the site exists!",
+      })
+    }
+
+    // Step 2: Retrieve review request
+    const possibleReviewRequest = await this.reviewRequestService.getReviewRequest(
+      site,
+      requestId
+    )
+
+    if (isIsomerError(possibleReviewRequest)) {
+      logger.error({
+        message: "Invalid review request requested",
+        method: "getBlob",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+          requestId,
+          file: path,
+        },
+      })
+      return res.status(404).send({
+        message: "Please ensure that the site exists!",
+      })
+    }
+
+    // Step 3: Check if the user is a contributor of the site
+    const role = await this.collaboratorsService.getRole(
+      siteName,
+      userWithSiteSessionData.isomerUserId
+    )
+
+    if (!role) {
+      logger.error({
+        message:
+          "User with insufficient permissions attempted to retrieve blob diff",
+        method: "getBlob",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+        },
+      })
+      return res.status(404).send({
+        message: "Please ensure that the site exists!",
+      })
+    }
+
+    // NOTE: Currently, Isomer only allows comparisons between staging and production.
+    // This might change in the future and in that case, the `getBlob` method call below
+    // should have the corresponding ref (`master` or `staging`) changed.
+    const prodPromise = await this.reviewRequestService.getBlob(
+      siteName,
+      path,
+      "master"
+    )
+    const stagingPromise = await this.reviewRequestService.getBlob(
+      siteName,
+      path,
+      "staging"
+    )
+
+    const data = await Promise.all([prodPromise, stagingPromise])
```

Comment:
Can I check: My understanding is that `prodPromise` and `stagingPromise` already has an `await`, which means that by this step, both values should have been resolved. Is this line still required?

Reply:
ah yes - i missed this. should've removed the `await` from earlier on, thanks for catching!